mod/wikirate_source/spec/set/type/source/source_spec.rb

```diff
@@ -2,58 +2,82 @@
 require 'link_thumbnailer'
 
 describe Card::Set::Type::Source do
-  describe "while creating a Source" do
+  def create_link_source url
+    create_source link: url
+  end
+
+  def create_source args
+    Card.create! source_args(args)
+  end
+
+  def source_args args
+    res = {
+      type_id: Card::SourceID,
+      subcards: {
+       '+Link' => {},
```

Comment:
Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.